### ISSUE TITLE
Add routing to chat example?

### ISSUE THREAD

**Issue by trungpham** [yahoo/flux-examples#72]
Would it be possible to add routing to the chat example. So when the user refreshes the page, it doesn't default to the last chat window?

**Comment by trungpham** [comment]
I'm working on this. Running into some issue with the navigation on the client side.

**Comment by timReynolds** [comment]
I've got this to work using the routr in the following branch. 
https://github.com/timReynolds/flux-examples/tree/feature.chatRoutr

However every click currently reloads the data from the store. Looking at the routr related components it appears that I should be able to match the event type e.g. pageload vs click. You can see I've tried adding the logic to the route so on click it only changes thread;

```
  thread_pageload: {
    path: '/thread/:id',
    method: 'get',
    navigate: {
      type: "pageload",
    },
    action: function (context, payload, done) {
      debug('thread pageload');
      context.executeAction(showChat, {}, function() {
        context.dispatch('OPEN_THREAD', { threadID: payload.params.id });
        done();
      });
    }
  },
  thread_click: {
    path: '/thread/:id',
    method: 'get',
    navigate: {
      type: "click",
    },
    action: function (context, payload, done) {
      debug('thread click');
      context.dispatch('OPEN_THREAD', { threadID: payload.params.id });
      done();
    }
  }
```

Can anyone give me some advice if this is possible and where I'm going wrong? From what I understand in routr this looked like it would try to match based on config.navigate at this point;

```
    // 3. check navParams, if this route has match requirements defined for navParams
    var navParamsConfig = (self.config.navigate && self.config.navigate.params);
```

However navParamsConfig comes back undefined although self.config.navigate.type is set to the correct value ("pagoda" or "click"). 

Thanks

**Comment by jedireza** [comment]
@timReynolds thanks. Your code looks pretty good. I think we should open a PR so we can do a code review. Just make the title is something like `[wip] Adding routing to chat example` so everyone knows it's a work in progress.

**Comment by timReynolds** [comment]
@jedireza will do. 

Still has a couple of issues as mentioned above. Routr is pretty undocumented however and didn't have time to dig through the code. Is there somewhere good to ask questions about the flux related yahoo libs?

**Comment by jedireza** [comment]
This is a good place to ask those questions. I'll have try running your branch when I get sometime.

**Comment by timReynolds** [comment]
@jedireza did you get a chance to take a look at the pull request? Would love to get this finished

**Comment by jedireza** [comment]
I found a solution and left some notes on the [PR](https://github.com/yahoo/flux-examples/pull/82).

**Comment by lingyan** [comment]
Adding navigation type to routr's match algorithm works, but that will make routes config bloat. 

An alternative and IMO better solution is the showChat action can be more intelligent to not fetch messages again if messages have already been fetched.  This way, it will work if there are other use cases where showChat is invoked directly.

**Comment by timReynolds** [comment]
Thanks for the feedback I've updated the pull request to include the logic into the showChat action.

**Comment by redonkulus** [comment]
Fixed with #82
